main.go

```diff
@@ -92,7 +92,15 @@ func main() {
 	})
 
 	svr.AddFunc(func(ctx context.Context) {
-		srv := server.New(ctx, ds, mq, rnr, tasks, server.DefaultEnqueue)
+		srv := server.New(
+			ctx,
+			ds,
+			mq,
+			rnr,
+			tasks,
+			server.DefaultEnqueue,
+			server.EnableShutdownEndpoint(halt),
```

Comment:
Is really necessary to add this to the server instance?
Can't this work almost like a plugin added in the main package?